src/JackTrip.cpp

```diff
@@ -158,15 +158,15 @@ void JackTrip::setupAudio(
 
 #ifdef WAIRTOHUB // WAIR
         //Set our Jack client name if we're a hub server or a custom name hasn't been set
-	      if ( mPeerAddress.toStdString() != "" &&
-	          (mJackClientName == gJackDefaultClientName || mJackTripMode == SERVERPINGSERVER)) {
-            mJackClientName = QString(mPeerAddress).replace(":", ".").toLatin1().constData();
+        if ( mPeerAddress.toStdString() != "" &&
+            (mJackClientName.constData() == gJackDefaultClientName.constData() || mJackTripMode == SERVERPINGSERVER)) {
+            mJackClientName = QString(mPeerAddress).replace(":", ".");
         }
 //        std::cout  << "WAIR ID " << ID << " jacktrip client name set to=" <<
 //                      mJackClientName << std::endl;
 #endif // endwhere
 
-        mAudioInterface->setClientName(mJackClientName);
+        mAudioInterface->setClientName(mJackClientName.toUtf8().constData());
```

Comment:
So I found out the hard way that this is a bad idea. The pointer returned by constData() is only guaranteed to be valid while the QByteArray returned by mJackClientName.toUtf8() still exists. And since we don't keep a reference to it, that isn't long. So we either need to keep a reference, or the better solution (and what I've done in the qjackctl branch) is to have the mClientName variable in JackAudioInterface be a QString and only make the toUtf8().constData() call when passing it as an argument to jack_client_open.

(It should be safe to use the version of JackAudioInterface.cpp and .h from the qjackctl branch because I don't think I've made any other crazy changes to them, but I'd need to double check that. From memory you'd also need to pull in AudioInterface.h as well.)

Reply:
It crossed my mind this could be a problem.
[Few lines above](https://github.com/jacktrip/jacktrip/pull/106/files#diff-67ee6d5b8fb8b9a04f9fea4980ac7d56R163) we also do  `mJackClientName = QString(mPeerAddress).replace(":", ".");`. Would that also need to change?

Reply:
No, that should be fine because we're keeping a reference to the result as a member variable.